<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="scottzijiezhang" />

<meta name="date" content="2018-12-01" />

<title>manual</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MeRIPtoolsManual</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="manual.html">Manual</a>
</li>
<li>
  <a href="workflow.html">Standard workflow</a>
</li>
<li>
  <a href="QTLflow.html">QTL discovery workflow</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/scottzijiezhang/MeRIPtools">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">manual</h1>
<h4 class="author"><em>scottzijiezhang</em></h4>
<h4 class="date"><em>2018-12-01</em></h4>

</div>


<div id="introduction" class="section level2">
<h2><strong>Introduction</strong></h2>
<p>Epitranscriptome profiling using MeRIP-seq is a powerful technique for in vivo functional studies of reversible RNA modifications. We developed MeRIPtools, a comprehensive tool to process and analyze aligned sequencing data. MeRIPtools also provide a framework to manage data associated with peak-calling, differential methylation analysis in R.</p>
</div>
<div id="data-structure" class="section level2">
<h2><strong>Data structure</strong></h2>
<div id="naming-convention-of-input-files" class="section level3">
<h3><strong>Naming convention of input files</strong></h3>
<p>Since MeRIP-seq generate a pair of sequencing data – <em>Input</em> and <em>IP</em>, we denote these two libraries by <code>.input</code> or <code>.m6A</code> followed by <code>.bam</code> after sample names. Thus, the RADAR requires a pair of BAM files for each sample named as <code>&lt;samplename&gt;.input.bam</code> + <code>&lt;samplename&gt;.&lt;modification&gt;.bam</code>. The <code>&lt;modification&gt;</code> can be specified in the <code>countRead(... , modification = &quot;&lt;modification&gt;&quot;)</code> function.<br />
All BAM files to be processed and analyzed together should be saved in one directory, the path to which needs to be specified in the <code>countRead(bamPath = &quot;.../.../...&quot;)</code>.</p>
</div>
<div id="the-merip-and-merip.peak-class" class="section level3">
<h3><strong>The MeRIP and MeRIP.Peak class</strong></h3>
<p>MeRIPtools used the S4 object system to manage the data involved in processing MeRIP-seq data. We defined a class <code>MeRIP</code> that stores the basic informations about a MeRIP-seq experiment (e.g. path to BAM files, read count matrix, gene model used to quantify reads etc.).<br />
We also defined two classes <code>MeRIP.Peak</code> and <code>MeRIP.RADAR</code> that inherits <code>MeRIP</code> class and each stores intermediate data and result involved in two different pipelines. <code>MeRIP.RADAR</code> is designed for peak-calling free differential methylation analysis pipeline as detailed in R package <a href="https://github.com/scottzijiezhang/RADAR"><em>RADAR</em></a>. <code>MeRIP.Peak</code> is designed for peak-calling based analysis pipeline, as implemented in <em>MeRIPtools</em>. We will focus on <code>MeRIP.Peak</code> class in this manual.</p>
<p>Data slots in <code>MeRIP</code> class:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getSlots</span>(<span class="st">&quot;MeRIP&quot;</span>)</code></pre></div>
<pre><code>##         reads       binSize     geneModel           gtf bamPath.input 
##      &quot;matrix&quot;     &quot;numeric&quot; &quot;GRangesList&quot;   &quot;character&quot;   &quot;character&quot; 
##    bamPath.ip   samplenames      geneBins       geneSum           GTF 
##   &quot;character&quot;   &quot;character&quot;  &quot;data.frame&quot;      &quot;matrix&quot;     &quot;GRanges&quot;</code></pre>
<p><code>reads</code> Numeric matrix data for read count in <em>input</em> and <em>IP</em> library. The columns are sample names and rows are consecutive bins of user defined size (defaul is 50bp).<br />
<code>binSize</code> Interger specifying the bin size used to divide the transcript.<br />
<code>geneModel</code> A GRangesList that stores the gene model that has been filtered for ambiguous chromosome. Each element of the list contains a GRanges object specifying exons of a gene.<br />
<code>gtf</code> The path to the gtf annotation file used to obtain gene model.<br />
<code>bamPath.input</code> The path to input BAM files.<br />
<code>bamPath.IP</code> The path to IP BAM files.<br />
<code>samplenames</code> The prefix used for naming <em>input</em> and <em>IP</em> BAM files (as introduced in naming convention), which specifies the name of a sample.<br />
<code>geneBins</code> Data.frame mapping each consecutive bins to the corresponding gene.<br />
<code>geneSum</code> Numeric matrix data for gene level read count in the <em>input</em> library. This is equivalent to the read count matrix commenly used in RNA-seq differential expression analysis.<br />
<code>GTF</code> The gtf annotation in GRanges format.</p>
<p>Data slots in <code>MeRIP.Peak</code> class:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getSlots</span>(<span class="st">&quot;MeRIP.Peak&quot;</span>)</code></pre></div>
<pre><code>##      peakCallResult  jointPeak_id_pairs          jointPeaks 
##            &quot;matrix&quot;            &quot;matrix&quot;        &quot;data.frame&quot; 
##        jointPeak_ip     jointPeak_input   norm.jointPeak_ip 
##            &quot;matrix&quot;            &quot;matrix&quot;            &quot;matrix&quot; 
##          sizeFactor             variate   jointPeak_adjExpr 
##        &quot;data.frame&quot;        &quot;data.frame&quot;            &quot;matrix&quot; 
##            test.est         peakCalling jointPeak_threshold 
##            &quot;matrix&quot;         &quot;character&quot;           &quot;numeric&quot; 
##         test.method               reads             binSize 
##         &quot;character&quot;            &quot;matrix&quot;           &quot;numeric&quot; 
##           geneModel                 gtf       bamPath.input 
##       &quot;GRangesList&quot;         &quot;character&quot;         &quot;character&quot; 
##          bamPath.ip         samplenames            geneBins 
##         &quot;character&quot;         &quot;character&quot;        &quot;data.frame&quot; 
##             geneSum                 GTF 
##            &quot;matrix&quot;           &quot;GRanges&quot;</code></pre>
<p><code>peakCallResult</code> Matrix data of logic data encode whether a bin (row) of a sample (column) is called a peak or not.<br />
<code>jointPeaks</code> For quantitative analysis, we are interested in the union of peaks across experimental groups, thus we define joint peak as requiring a bin to be called a peak in at least <jointPeak_threshold> of samples. This data.frame stores the joint peak reported at user defined <code>jointPeak_threshold</code> in BED12 format.<br />
<code>test.est</code> Parameter estimation by differential test either by RADARtest (PoissonGamma test), QNB (quad-negative binomial test) or Beta-binomial test.<br />
<code>jointPeak_threshold</code> The threshold used to define joint peak in current MeRIP.Peak object.<br />
<code>test.method</code> The statistical method used to make inferential test.<br />
<code>variate</code> This data.frame store the predictor variable (the first column) used for inferential test and covariates (the rest of columns) to be included.<br />
<code>sizeFactor</code> The estimated size factor for both Input and IP.</p>
</div>
</div>
<div id="functions" class="section level2">
<h2><strong>Functions</strong></h2>
<div id="countreads" class="section level3">
<h3>countReads</h3>
<p>This is the very first function in MeRIP-seq data analysis that initianize a <code>MeRIP</code> object. This function takes BAM files of Input/IP library of each samples as input and use given GTF file as gene annotation to divide genes into consecutive bins of user defined size.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">countReads</span>(
  samplenames,
  gtf, 
  <span class="dt">fragmentLength =</span> <span class="dv">150</span>,
  bamFolder,
  <span class="dt">outputDir=</span><span class="ot">NA</span>,
  <span class="dt">modification =</span> <span class="st">&quot;m6A&quot;</span>,
  <span class="dt">binSize =</span> <span class="dv">50</span>,
  <span class="dt">strandToKeep =</span> <span class="st">&quot;opposite&quot;</span>,
  <span class="dt">paired =</span> <span class="ot">FALSE</span>,
  <span class="dt">threads =</span> <span class="dv">1</span>,
  <span class="dt">saveOutput =</span> T
)</code></pre></div>
<p>Input parameters: The <code>samplenames</code> should be a vector of charaters that are the same as the prefix name of input BAM files. BAM files should be named as <code>&lt;samplename&gt;.input.bam</code> and <code>&lt;samplename&gt;.&lt;modification&gt;.bam</code>.<br />
The <code>gtf</code> shoudl be a (charater) path to the gft annotation file.<br />
The <code>fragmentLength</code> is the average length of RNA fragment. Default is 150bp.<br />
The <code>bamFolder</code> should be a (charater) path to the dirctory where input BAM files are located.<br />
The <code>outputDir</code> denote the path to save the read count result in RDS format.<br />
The <code>modification</code> is a character denoting the mid-name of BAM files for IP sample. Default is “m6A”, where <code>countReads</code> function expect to find <code>&lt;samplenames&gt;.m6A.bam</code> in the <code>bamPath</code> directory.<br />
The <code>binSize</code> is an integer indicating the size of bin to divide the transcript.<br />
The <code>strandToKeep</code> parameter set the strand specificity when counting the reads. Default is “opposite” for reverse-stranded library construction protocol, which is the most common one seen in RNA-seq. “same” denotes the sense-stranded library construction protocol, commonly seen in small RNA library kit. This parameter can also be set to “both” to ignore strand when counting the reads.<br />
The <code>paired</code> is a logic parameter. Default is FALSE for single-end sequencing data. TURE for pair-end sequencing data where fragment length is calculated from the read pair instead of using the user provided parameter.<br />
The <code>threads</code> set the number of threads to use for multi-core processing.<br />
The <code>saveOutput</code> is logic parameter denoting whether read count result should be saved.</p>
<p>This function will return a MeRIP class object that stores the read count in bins for each samples.</p>
</div>
<div id="callpeakfisher" class="section level3">
<h3>callPeakFisher</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">callPeakFisher</span>(<span class="dt">MeRIP =</span> MeRIP_object,<span class="dt">min_counts =</span> <span class="dv">15</span>, <span class="dt">peak_cutoff_fdr =</span> <span class="fl">0.05</span>, <span class="dt">peak_cutoff_oddRatio =</span> <span class="dv">1</span>, <span class="dt">threads =</span> <span class="dv">1</span>)</code></pre></div>
</div>
<div id="callpeakbinomial" class="section level3">
<h3>callPeakBinomial</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">callPeakBinomial</span>(<span class="dt">MeRIP =</span> MeRIP_object,<span class="dt">min_counts =</span> <span class="dv">15</span>, <span class="dt">peak_cutoff_fdr =</span> <span class="fl">0.05</span>, <span class="dt">peak_cutoff_oddRatio =</span> <span class="dv">1</span>, <span class="dt">threads =</span> <span class="dv">1</span>)</code></pre></div>
</div>
</div>
<div id="reference" class="section level2">
<h2><strong>Reference</strong></h2>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
