<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="scottzijiezhang" />

<meta name="date" content="2018-12-01" />

<title>Example work flow</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MeRIPtoolsManual</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="manual.html">Manual</a>
</li>
<li>
  <a href="workflow.html">Standard workflow</a>
</li>
<li>
  <a href="QTLflow.html">QTL discovery workflow</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/scottzijiezhang/MeRIPtools">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Example work flow</h1>
<h4 class="author"><em>scottzijiezhang</em></h4>
<h4 class="date"><em>2018-12-01</em></h4>

</div>


<div id="quick-start" class="section level2">
<h2>1.Quick start</h2>
<p>Here we show the most basic steps for a differential methylation analysis. There is one major step upstream of current analysis that generates mapped reads in BAM files for each sample, which can be done by varies choices of softwares such as <a href="https://ccb.jhu.edu/software/hisat2/index.shtml">Hisat2</a>, <a href="http://www.ccb.jhu.edu/software/tophat/index.shtml">tophat2</a> or <a href="https://pachterlab.github.io/kallisto/">Kallisto</a>. we will discuss in the sections below assuming <strong>BAM</strong> file has been obtained.</p>
<p>This code chunk assumes that you have bam files for sample1-6 <code>s1.input.bam... s6..input.bam</code> and <code>s1.m6A.bam... s6..m6A.bam</code>.</p>
<p>lazy pipeline not implemented yet…</p>
</div>
<div id="standard-workflow" class="section level2">
<h2>2.Standard workflow</h2>
<div id="a.-mapped-bam-file." class="section level3">
<h3>a). Mapped BAM file.</h3>
<p>To analyze for MeRIP-seq data, <strong>MeRIPtools</strong> expect a pair of BAM file for each sample. For each pair of BAM file, the naming convention for <strong>MeRIPtools</strong> is <code>sample.input.bam</code> for <strong>INPUT</strong> and <code>sample.m6A.bam</code> for m6A <strong>IP</strong> or <code>sample.m1A.bam</code> for m1A <strong>IP</strong> sample. They key is <strong>INPUT</strong> and <strong>IP</strong> sample have the same prefix. The mid-name for <strong>IP</strong> sample can be specified in the <code>countReads</code> function and thus can be anything.</p>
</div>
<div id="b.-gtf-annotation-file." class="section level3">
<h3>b). GTF annotation file.</h3>
<p>For the organism being analyzed, user need to provide an annotation file in gtf format to define the genomic coordinate of gene features. A good source to download those supporting files are <a href="https://support.illumina.com/sequencing/sequencing_software/igenome.html">iGenome</a>. Alternatively, the <a href="https://www.gencodegenes.org/">GENECODE</a> is also a good source for annotation if you are interested in human and mouse genome.</p>
</div>
<div id="c.-the-merip-class" class="section level3">
<h3>c). The MeRIP class</h3>
<p>The MeRIPtools uses a S4 class system to manage the data involved in the analysis, please see <a href="">Manual</a> page for more details. The <code>countReads()</code> function takes BAM files and gtf annotation as inputs and initializes a MeRIP class object by counting the reads in consecutive bins of size set by the user.</p>
<p>First, let’s prepare input data.</p>
<pre class="r"><code>library(MeRIPtools)
samplenames &lt;- c(&quot;4582WT&quot;,&quot;4584WT&quot;,&quot;4587WT&quot;,&quot;4612WT&quot;,&quot;4583KO&quot;,&quot;4586KO&quot;,&quot;4615KO&quot;,&quot;4616KO&quot;)
bam_dir &lt;- &quot;~/METTL14_liver2017/bam_files&quot;</code></pre>
<p>The variable <code>samplenames</code> correspond to the prefix of file names in the <code>bam_dir</code>. We can check the files in the <code>bam_dir</code>:</p>
<pre class="r"><code>list.files(bam_dir)</code></pre>
<pre><code>##  [1] &quot;4582WT.input.bam&quot;     &quot;4582WT.input.bam.bai&quot; &quot;4582WT.m6A.bam&quot;      
##  [4] &quot;4582WT.m6A.bam.bai&quot;   &quot;4583KO.input.bam&quot;     &quot;4583KO.input.bam.bai&quot;
##  [7] &quot;4583KO.m6A.bam&quot;       &quot;4583KO.m6A.bam.bai&quot;   &quot;4584WT.input.bam&quot;    
## [10] &quot;4584WT.input.bam.bai&quot; &quot;4584WT.m6A.bam&quot;       &quot;4584WT.m6A.bam.bai&quot;  
## [13] &quot;4586KO.input.bam&quot;     &quot;4586KO.input.bam.bai&quot; &quot;4586KO.m6A.bam&quot;      
## [16] &quot;4586KO.m6A.bam.bai&quot;   &quot;4587WT.input.bam&quot;     &quot;4587WT.input.bam.bai&quot;
## [19] &quot;4587WT.m6A.bam&quot;       &quot;4587WT.m6A.bam.bai&quot;   &quot;4612WT.input.bam&quot;    
## [22] &quot;4612WT.input.bam.bai&quot; &quot;4612WT.m6A.bam&quot;       &quot;4612WT.m6A.bam.bai&quot;  
## [25] &quot;4615KO.input.bam&quot;     &quot;4615KO.input.bam.bai&quot; &quot;4615KO.m6A.bam&quot;      
## [28] &quot;4615KO.m6A.bam.bai&quot;   &quot;4616KO.input.bam&quot;     &quot;4616KO.input.bam.bai&quot;
## [31] &quot;4616KO.m6A.bam&quot;       &quot;4616KO.m6A.bam.bai&quot;</code></pre>
<p>It is okay if your folder don’t have <code>xxx.bai</code> files. These are BAM file index, which can be generated by the MeRIPtools.</p>
<p>The next step is to count reads in consecutive bins. Usually we do 50 bp bins for typical library of 20-30 Million reads. If you have deeper coverage, a smaller bin size could increase resolution of the analysis. However, for libraries of shallow coverage, larger bin size is recommended because the smaller the bin size, the less reads are sampled in a bin and therefore the larger sampling error is encountered.</p>
<pre class="r"><code>MeRIP &lt;- countReads(samplenames = samplenames,
                    gtf = &quot;~/Database/genome/GRCm38.p6/gencode.vM18.annotation.gtf&quot;,
                    bamFolder = bam_dir,
                    outputDir = NA,
                    modification = &quot;m6A&quot;,
                    strandToKeep = &quot;opposite&quot;,
                    fragmentLength = 150,
                    binSize = 50,
                    threads = 20,
                    saveOutput = FALSE
                    )</code></pre>
<p><strong>Note 1</strong> The parameter <code>strandToKeep</code> defines the strand of read to count with respect to the transcript strand. Usually, Illumina Truseq library kit or other similar protocol resulted in sequencing opposite strand of the RNA molecules in read1; therefore, default is “opposite” for this parameter. However, for small RNA library protocol, the sense strand of the RNA molecule is sequenced as read1; thus “same” should be set to this parameter.<br />
<strong>Note 2</strong> The parameter <code>fragmentLength</code> defines the average length of the RNAs (insert size of the library). This parameter will affect the location of the reads to be counted. This parameter can be learned from data for paired-end sequencing data.</p>
<p>A summary of the MeRIP object can be printed by just call the variable name (“MeRIP” in this example). Or use the <code>summary(MeRIP)</code> function.</p>
<pre class="r"><code>MeRIP</code></pre>
<pre><code>## MeRIP dataset of 8 samples.
## The total read count for Input and IP samples are (Million reads):
##       4582WT 4584WT 4587WT 4612WT 4583KO 4586KO 4615KO 4616KO
## Input  18.30  15.90  16.00  16.35  20.18  16.77  15.47  17.81
## IP     18.67  21.54  19.83  20.06  18.82  21.43  20.44  18.06
## 
## Peak calling done by Binomial test.
## 40186 joint peak reported at threshold 2 (requiring peaks to be called in at least 2 samples).
## Input gene level count available.
## There are 1 predictor variables/covariates. Can access by function variable(MeRIPdata). 
## [1] &quot;Differential peaks tested by QNB test.\n&quot;</code></pre>
</div>
<div id="d.-perform-peak-calling-for-each-sample." class="section level3">
<h3>d). Perform peak calling for each sample.</h3>
<p>After obtaining read count in consecutive bins, we can perform peak calling on each bin of each sample. This step will result in a matrix of logic value to indicate whether each bin is called a peak or not. The result will be stored in a MeRIP.Peak object and returned.<br />
Currently, two methods were implemented in MeRIPtools to call peak.</p>
<p><strong>(1) Binomial test</strong> This method models the total read count (Input + IP) and expected ratio of read count in IP as binomial distribution and test the probability of observing the total read count and IP read count where expected ratio <span class="math inline">\(p\)</span> is given by the library size (total coverage) of IP/(Input+IP).</p>
<pre class="r"><code>MeRIP &lt;- callPeakBinomial(MeRIP = MeRIP, min_counts = 15, peak_cutoff_fdr = 0.05, peak_cutoff_oddRatio = 1, threads = 20)</code></pre>
<p>The parameter <code>min_counts</code> set the minimal number of total reads (Input + IP) presented in a bin to call this bin peak. The parameter <code>peak_cutoff_fdr</code> set the FDR threshold (correct for multiple tests for bins in each gene) to call a bin peak. The parameter <code>peak_cutoff_oddRatio</code> set the IP/Input ratio threshold to call a bin peak.</p>
<p><strong>(2) Fisher’s exact test</strong> This method test a contingency table of read count in IP and gene-wise median read count in IP as well as read count in Input and gene-wise median read count in Input. Basically, it test for the deviation of read count distribution in IP compare to expectation given by the Input sample. This method assumes that only a small segment of a gene is enriched by the IP experiment as gene-wise median was used as a control. When the whole is enriched (e.g. very short transcript or widely methylated genes), this test would fail to detect enrichment.</p>
<pre class="r"><code>MeRIP &lt;- callPeakFisher(MeRIP = MeRIP, min_counts = 15, peak_cutoff_fdr = 0.05, peak_cutoff_oddRatio = 1, threads = 20)</code></pre>
<p>The parameters are the same as previous method except for the oddRatio, which is represented by the (IP/gene-median-IP)/(Input/gene-median-Input).</p>
</div>
<div id="e.-define-joint-peak-for-quantitative-analysis." class="section level3">
<h3>e). Define joint peak for quantitative analysis.</h3>
<p>Although peak calling has been perform for each samples and there are variations of peaks called across samples. However, it is <strong>not</strong> a good idea to directly compare across samples (groups) using the peak calling result (qualitative info) to infer changes of methylation status between groups/conditions (e.g. define emerged/lost peaks). There are two major drawbacks of such analysis: (1) some peak calling result variation actually reflect boundary cases where peak is called in one group with FDR = 0.049 while peak will <strong>not</strong> be called in another group with FDR = 0.051. However, the methylation level are not really different in these two group as the emerged/lost peak seemingly suggested. (2) peak could be call in one group but not another due to differential gene expression. Low expression of a gene in one group could resulted in total read counts lower than the minimal required read count for peak calling and thus not called as a peak. However the IP/Input ratio of this groupd could be roughly the same as another group.<br />
Thus, we recommand using quantitative analysis to search for differential methylated regions (peaks). On one hand, we are interested in peaks that are vary across samples, thus we want to make differential test on the union of peaks. On another hand, we don’t want to include peaks that are called only one but not any other samples because these peaks are likely false signals. Thus, we defined a joint peak that requires peak to be called in at least 2 (or any other threshold the user defines) samples for comparative analysis. The joint peaks is an representation of the union of peaks in samples analyzed.</p>
<pre class="r"><code>MeRIP &lt;- reportJointPeak(MeRIPdata = MeRIP, joint_threshold = 2, threads = 10)</code></pre>
<p>Now we have joint peaks stored in the MeRIP.Peak object returned from <code>reportJointPeak()</code> function. We can get the joint Peak by <code>jointPeaks(MeRIP)</code>.</p>
<pre class="r"><code>head( jointPeak(MeRIP) )</code></pre>
<pre><code>##     chr     start       end                  name score strand thickStart
## 1  chr3 108108911 108109157  ENSMUSG00000000001.4     0      -  108108911
## 2  chr3 108109207 108109256  ENSMUSG00000000001.4     0      -  108109207
## 3  chr3 108123800 108146097  ENSMUSG00000000001.4     0      -  108123800
## 4  chr7 142577946 142577995 ENSMUSG00000000031.16     0      -  142577946
## 5 chr11 121237303 121242550  ENSMUSG00000000056.7     0      +  121237303
## 6 chr11 121252834 121253082  ENSMUSG00000000056.7     0      +  121252834
##    thickEnd itemRgb blockCount blockSizes blockStarts
## 1 108109157       0          1        246           0
## 2 108109256       0          1         49           0
## 3 108146097       0          2     38,210     0,22088
## 4 142577995       0          1         49           0
## 5 121242550       0          3   49,99,50 0,1116,5198
## 6 121253082       0          1        248           0</code></pre>
</div>
<div id="f.-extract-read-count-of-joint-peaks" class="section level3">
<h3>f). Extract read count of joint Peaks</h3>
<p>After obtaining the joint peaks, we will need to summarize the read count in peaks for Input and IP.</p>
<pre class="r"><code>MeRIP &lt;- jointPeakCount(MeRIP)</code></pre>
<p>This step is to prepare read count data for differential methylation test. If you are interested in peak read count data, you can extract it by:</p>
<pre class="r"><code>Input_count &lt;- extractInput(MeRIP)
IP_count &lt;- extractIP(MeRIP)

dim(Input_count)</code></pre>
<pre><code>## [1] 40186     8</code></pre>
</div>
<div id="g.-inferential-test" class="section level3">
<h3>g). Inferential test</h3>
<p>Usually, we are interested in compare samples across experiment conditions or genotypes. We implemented three alternative tests in MeRIPtools: <code>RADARtest</code> using a random effect model, <code>QNBtest</code> using quad-negative binomial model and <code>BetaBinTest</code> using beta-binomial regression model. Before we run the test, we need to set the variable associated with each samples. Predictor variable and potential covariates are inputed as a data.frame.</p>
<pre class="r"><code>variable(MeRIP) &lt;- data.frame( genotype = c(&quot;WT&quot;,&quot;WT&quot;,&quot;WT&quot;,&quot;WT&quot;,&quot;KO&quot;,&quot;KO&quot;,&quot;KO&quot;,&quot;KO&quot;) )

## if there is covariates
variable(MeRIP) &lt;- data.frame( genotype = c(&quot;WT&quot;,&quot;WT&quot;,&quot;WT&quot;,&quot;WT&quot;,&quot;KO&quot;,&quot;KO&quot;,&quot;KO&quot;,&quot;KO&quot;), covariate1 = c(...),  covariate2 = c(...), ... )</code></pre>
<div id="qnb" class="section level4">
<h4><strong>(1) QNB</strong></h4>
<p>QNB test was introduced by <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1808-4">Liu et al 2017 <em>BMC Bioinformatics</em></a>. Here we implemented a wrapper function to call QNB test. <strong>Note</strong> this test is designed for small sample size scenario and cannot incorporate covariates. Thus I recommond using this test for cell line experiment where only 2-3 technical/biological replicates are available.</p>
<pre class="r"><code>MeRIP &lt;- QNBtest(MeRIP)</code></pre>
</div>
<div id="radartest" class="section level4">
<h4><strong>(2) RADARTest</strong></h4>
<p>To be add… #### <strong>(3) BetaBinTest</strong> To be add…</p>
</div>
</div>
<div id="h.-check-result" class="section level3">
<h3>h). Check result</h3>
<p>Finally, we want to check the joint peaks and associated test statistics. The function <code>results()</code> will extract the joint peaks and associated differential test statistics.</p>
<pre class="r"><code>allPeakResult &lt;- results(MeRIP)

head(allPeakResult)</code></pre>
<pre><code>##     chr     start       end                  name score strand thickStart
## 1  chr3 108108911 108109157  ENSMUSG00000000001.4     0      -  108108911
## 2  chr3 108109207 108109256  ENSMUSG00000000001.4     0      -  108109207
## 3  chr3 108123800 108146097  ENSMUSG00000000001.4     0      -  108123800
## 4  chr7 142577946 142577995 ENSMUSG00000000031.16     0      -  142577946
## 5 chr11 121237303 121242550  ENSMUSG00000000056.7     0      +  121237303
## 6 chr11 121252834 121253082  ENSMUSG00000000056.7     0      +  121252834
##    thickEnd itemRgb blockCount blockSizes blockStarts p.treated p.control
## 1 108109157       0          1        246           0 0.6107835 0.7072817
## 2 108109256       0          1         49           0 0.1892661 0.2908680
## 3 108146097       0          2     38,210     0,22088 0.4098838 0.3320385
## 4 142577995       0          1         49           0 0.2791100 0.5723506
## 5 121242550       0          3   49,99,50 0,1116,5198 0.4373165 0.6007194
## 6 121253082       0          1        248           0 0.2459933 0.4964980
##      log2.RR       beta     p_value         q      padj
## 1 -0.2116238 -0.6226838 0.107213503 733.70257 0.9026464
## 2 -0.6199482 -0.8131224 0.480945117  59.62527 0.9895912
## 3  0.3038641  0.4826298 0.383857934 297.92652 0.9895912
## 4 -1.0360652 -1.7894158 0.399415869  11.77369 0.9895912
## 5 -0.4580136 -0.9529343 0.132806105 193.58893 0.9466131
## 6 -1.0131691 -1.5957490 0.007498819 345.49632 0.4312254</code></pre>
<p>The column <code>beta</code> is the coefficient associated with the predictor variable. The interpretation is log2-fold change in <code>QNBtest</code> (log fold change RADARtest and BetaBinTest). The <code>p_value</code> is the P value for the differential test of each peak. <code>padj</code> is P value adjusted for multiple testing, which can be interpreted as false discovery rate (FDR).</p>
<p>To filter for significantly differentially methylated peaks, we usually use a FDR &lt; 10% cutoff to filter the result:</p>
<pre class="r"><code>diff_peaks &lt;- allPeakResult[allPeakResult$padj &lt; 0.1,]

dim(diff_peaks)</code></pre>
<pre><code>## [1] 688  19</code></pre>
<p>There are 688 peaks differentially methylated peaks tested by QNB in this example. You can save the result to a file by:</p>
<pre class="r"><code>write.table(diff_peaks, &quot;Path/to/file/location.xls&quot;, sep = &quot;\t&quot;, col.names = T, row.names = F, quote = F)</code></pre>
</div>
</div>
<div id="peak-calling-only-analysis" class="section level2">
<h2>3. Peak calling only analysis</h2>
<p>If you are only interested in peak calling and want to obtain peaks consistent across some samples, you can report consistent peak after step <strong>2 d)</strong> in previous section.</p>
<p>For example, I’m only interested in peaks that are consistent in the first two samples:</p>
<pre class="r"><code>consistentPeak1 &lt;- consistentPeak(MeRIP, samplenames = c(&quot;4582WT&quot;,&quot;4584WT&quot;), joint_threshold = 2)</code></pre>
<p>Or I can report peaks that are consistent in at least 3 samples among 4 samples indicated by setting the <code>joint_threshold=3</code> :</p>
<pre class="r"><code>consistentPeak2 &lt;- consistentPeak(MeRIP, samplenames = c(&quot;4582WT&quot;,&quot;4584WT&quot;,&quot;4587WT&quot;,&quot;4612WT&quot;), joint_threshold = 3 )</code></pre>
<p><strong>Warning</strong> This function is intended to quality check peaks called in subset of samples or to just report a peak file. It is strongly not recommended to use this function to report peaks for two groups and compare the two peak files to infer variation between groups.</p>
</div>
<div id="session-info" class="section level2">
<h2>Session Info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.5.0 (2018-04-23)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 17.10
## 
## Matrix products: default
## BLAS: /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
##  [1] grid      stats4    parallel  stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] MeRIPtools_0.1.0            qvalue_2.14.0              
##  [3] RcppArmadillo_0.9.200.5.0   Rcpp_1.0.0                 
##  [5] reshape2_1.4.3              GenomicAlignments_1.18.0   
##  [7] SummarizedExperiment_1.12.0 DelayedArray_0.8.0         
##  [9] BiocParallel_1.16.2         matrixStats_0.54.0         
## [11] rtracklayer_1.42.1          doParallel_1.0.14          
## [13] iterators_1.0.10            foreach_1.4.4              
## [15] ggplot2_3.1.0               Rsamtools_1.34.0           
## [17] Biostrings_2.50.1           XVector_0.22.0             
## [19] GenomicFeatures_1.34.1      AnnotationDbi_1.44.0       
## [21] Biobase_2.42.0              GenomicRanges_1.34.0       
## [23] GenomeInfoDb_1.18.1         IRanges_2.16.0             
## [25] S4Vectors_0.20.1            BiocGenerics_0.28.0        
## 
## loaded via a namespace (and not attached):
##  [1] colorspace_1.3-2       rprojroot_1.3-2        htmlTable_1.12        
##  [4] base64enc_0.1-3        rstudioapi_0.8         QNB_1.1.11            
##  [7] bit64_0.9-7            codetools_0.2-15       splines_3.5.0         
## [10] R.methodsS3_1.7.1      geneplotter_1.60.0     knitr_1.20            
## [13] Formula_1.2-3          workflowr_1.1.1        broom_0.5.0           
## [16] annotate_1.60.0        cluster_2.0.7-1        R.oo_1.22.0           
## [19] gamlss.dist_5.1-0      compiler_3.5.0         httr_1.3.1            
## [22] backports_1.1.2        assertthat_0.2.0       Matrix_1.2-15         
## [25] lazyeval_0.2.1         acepack_1.4.1          htmltools_0.3.6       
## [28] prettyunits_1.0.2      tools_3.5.0            bindrcpp_0.2.2        
## [31] gtable_0.2.0           glue_1.3.0             GenomeInfoDbData_1.2.0
## [34] dplyr_0.7.8            ape_5.2                nlme_3.1-137          
## [37] pinfsc50_1.1.0         stringr_1.3.1          XML_3.98-1.16         
## [40] zlibbioc_1.28.0        MASS_7.3-51.1          scales_1.0.0          
## [43] BSgenome_1.50.0        hms_0.4.2              gamlss.data_5.1-0     
## [46] RColorBrewer_1.1-2     yaml_2.2.0             memoise_1.1.0         
## [49] gridExtra_2.3          biomaRt_2.38.0         rpart_4.1-13          
## [52] latticeExtra_0.6-28    stringi_1.2.4          RSQLite_2.1.1         
## [55] genefilter_1.64.0      permute_0.9-4          checkmate_1.8.5       
## [58] exomePeak_2.16.0       rlang_0.3.0.1          pkgconfig_2.0.2       
## [61] bitops_1.0-6           evaluate_0.12          lattice_0.20-38       
## [64] purrr_0.2.5            bindr_0.1.1            Guitar_1.20.0         
## [67] htmlwidgets_1.3        bit_1.1-14             tidyselect_0.2.5      
## [70] plyr_1.8.4             magrittr_1.5           DESeq2_1.22.1         
## [73] R6_2.3.0               Hmisc_4.1-1            DBI_1.0.0             
## [76] mgcv_1.8-26            pillar_1.3.0           foreign_0.8-71        
## [79] withr_2.1.2            survival_2.43-3        RCurl_1.95-4.11       
## [82] nnet_7.3-12            tibble_1.4.2           crayon_1.3.4          
## [85] gamlss_5.1-2           vcfR_1.8.0             rmarkdown_1.10        
## [88] progress_1.2.0         locfit_1.5-9.1         data.table_1.11.8     
## [91] vegan_2.5-3            blob_1.1.1             digest_0.6.18         
## [94] xtable_1.8-3           tidyr_0.8.2            R.utils_2.7.0         
## [97] munsell_0.5.0          viridisLite_0.3.0</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
